### PR TITLE
chore: set logger level to error to reduce noise from Elasticsearch and OpenSearch client libraries

### PR DESCRIPTION
**Pull request by jfcalvo**
# Description

This PR includes some changes to reduce the logger level of Elasticsearch and OpenSearch to error. With this we are reducing a little bit the output noise of these client libraries.

I have improved a little bit the settings so only `elasticsearch` and `opensearch` are valid options for `search_engine` attribute. I have added some small useful functions too.

Refs #4946 

**Type of change**

(Please delete options that are not relevant. Remember to title the PR according to the type of change)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (change restructuring the codebase without changing functionality)
- [x] Improvement (change adding some improvement to an existing functionality)
- [ ] Documentation update

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes. And ideally, reference `tests`)

- [x] Manually testing that warning messages are not showed anymore using Elasticsearch and OpenSearch.

**Checklist**

- [ ] I added relevant documentation
- [ ] follows the style guidelines of this project
- [ ] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK) (see text above)
- [ ] I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

File: argilla-server/src/argilla_server/settings.py [hunk 1]
```diff
@@ -97,7 +99,10 @@ class Settings(BaseSettings):
 
     es_mapping_total_fields_limit: int = 2000
 
-    search_engine: str = "elasticsearch"
+    search_engine: Union[
+        Literal[SEARCH_ENGINE_ELASTICSEARCH],
+        Literal[SEARCH_ENGINE_OPENSEARCH],
+    ] = SEARCH_ENGINE_ELASTICSEARCH
```
Comment (frascuchon) [inline argilla-server/src/argilla_server/settings.py]: Even if we can define constants for official engines, we shouldn't fix the available options. 

Creating new custom engine or specific engine connection could be helpful for the community but, by fixing this we’re avoiding this possibility. (See how engine [are registered](https://github.com/argilla-io/argilla/blob/chore/set-logger-level-to-error-for-search-engine/argilla-server/src/argilla_server/search_engine/elasticsearch.py#L40))

File: argilla-server/src/argilla_server/settings.py [hunk 2]
```diff
@@ -217,6 +222,21 @@ def old_dataset_records_index_name(self) -> str:
...
+    @property
+    def search_engine_is_elasticsearch(self) -> bool:
+        return self.search_engine == SEARCH_ENGINE_ELASTICSEARCH
+
+    @property
+    def search_engine_is_opensearch(self) -> bool:
+        return self.search_engine == SEARCH_ENGINE_OPENSEARCH
+
+    @property
+    def humanized_search_engine(self) -> str:
```
Comment (frascuchon) [inline argilla-server/src/argilla_server/settings.py]: The name of this method gives me chills...

"I've seen things you people wouldn't believe. Attack ships on fire off the shoulder of Orion. I watched C-beams glitter in the dark near the Tannhäuser Gate. All those moments will be lost in time like tears in rain..." 

Could we just call `human_readable_search_engine`? And also generalize a bit by using str.title() ? I think we don't need al this extra functions if we keep simple

**Review by frascuchon — CHANGES_REQUESTED** [review]

File: argilla-server/src/argilla_server/settings.py (hunk 2 above)
Comment (jfcalvo) [reply argilla-server/src/argilla_server/settings.py]: "Humanize" is a very common concept and naming when formatting strings but I don't have any problem changing it to "human_readable_*". 🙂

Some examples:
* One Python library: https://python-humanize.readthedocs.io/en/latest/
* A Django module: https://docs.djangoproject.com/en/5.0/ref/contrib/humanize/
* Moment.js classic JS library: https://momentjs.com/docs/#/durations/humanize/
* Size: an open source library that I made some years ago https://github.com/jfcalvo/size?tab=readme-ov-file#humanize-file-sizes

File: argilla-server/src/argilla_server/settings.py (hunk 1 above)
Comment (jfcalvo) [reply argilla-server/src/argilla_server/settings.py]: But in the case that a PR from the community adds a new engine they should modify this `Union` as part of the PR right? I don't get the problem.

> (See how engine [are registered](https://github.com/argilla-io/argilla/blob/chore/set-logger-level-to-error-for-search-engine/argilla-server/src/argilla_server/search_engine/elasticsearch.py#L40))

I missed that one. I will use the constant there too.

**Comment by jfcalvo** [comment]
@frascuchon I applied feedback from our discussion:
* Removed the `humanized_search_engine` property from settings.
* Removed the `Union` restricting the search_engine only to `elasticsearch` or `opensearch` values.

**Review by frascuchon — APPROVED** [review]

**Pull request merged**